services/secondary-storage/controller/src/main/java/org/apache/cloudstack/secondarystorage/SecondaryStorageManagerImpl.java

```diff
@@ -402,11 +402,17 @@ private List<String> getAllowedInternalSiteCidrs() {
         }
         String[] cidrs = _allowedInternalSites.split(",");
         for (String cidr : cidrs) {
-            if (NetUtils.isValidIp4Cidr(cidr) || NetUtils.isValidIp4(cidr) || !cidr.startsWith("0.0.0.0")) {
-                if (NetUtils.getCleanIp4Cidr(cidr).equals(cidr)) {
+            if (NetUtils.isValidIp4Cidr(cidr)) {
+                if (! NetUtils.getCleanIp4Cidr(cidr).equals(cidr)) {
                     s_logger.warn(String.format("Invalid CIDR %s in %s", cidr, SecStorageAllowedInternalDownloadSites.key()));
                 }
-                allowedCidrs.add(cidr);
+                allowedCidrs.add(NetUtils.getCleanIp4Cidr(cidr));
+            } else if (NetUtils.isValidIp4(cidr)) {
+                String newCidr = cidr + "/32";
```

Comment:
this is good.

alternatively we can pass `cidr`.
```
allowedCidrs.add(cidr);
```

the results are same
```
ip route add <ip> via <gw>
ip route add <ip>/32 via <gw>
```

Reply:
which one of the below changes would you prefer @weizhouapache

Reply:
both are fine I believe